server/service/src/sync/translations/barcode.rs

```diff
@@ -0,0 +1,136 @@
+use repository::{
+    BarcodeRow, BarcodeRowRepository, ChangelogRow, ChangelogTableName, StorageConnection,
+    SyncBufferRow,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::sync::api::RemoteSyncRecordV5;
+
+use super::{IntegrationRecords, LegacyTableName, PullUpsertRecord, SyncTranslation};
+
+const LEGACY_TABLE_NAME: &'static str = LegacyTableName::BARCODE;
+
+fn match_pull_table(sync_record: &SyncBufferRow) -> bool {
+    sync_record.table_name == LEGACY_TABLE_NAME
+}
+fn match_push_table(changelog: &ChangelogRow) -> bool {
+    changelog.table_name == ChangelogTableName::Barcode
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize)]
+pub struct LegacyBarcodeRow {
+    #[serde(rename = "ID")]
+    pub id: String,
+    #[serde(rename = "barcode")]
+    pub value: String,
+    #[serde(rename = "itemID")]
+    pub item_id: Option<String>,
+    #[serde(rename = "manufacturerID")]
+    pub manufacturer_id: Option<String>,
+    #[serde(rename = "packSize")]
+    pub pack_size: Option<i32>,
+    #[serde(rename = "parentID")]
+    pub parent_id: Option<String>,
+}
+
+pub(crate) struct BarcodeTranslation {}
+impl SyncTranslation for BarcodeTranslation {
+    fn try_translate_pull_upsert(
+        &self,
+        _: &StorageConnection,
+        sync_record: &SyncBufferRow,
+    ) -> Result<Option<IntegrationRecords>, anyhow::Error> {
+        if !match_pull_table(sync_record) {
+            return Ok(None);
+        }
+
+        let deserialised_row = serde_json::from_str::<LegacyBarcodeRow>(&sync_record.data);
+        if let Err(e) = deserialised_row {
```

Comment:
this is because I screw up the JSON every time I write these things - and the error from serde is quietly swallowed and the test fails. Warning like this allows me to see what I've done wrong when running the test.